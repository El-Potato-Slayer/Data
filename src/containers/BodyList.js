import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function BodyList(props) {
  const { bodyType } = props;
  let { bodies } = props;
  if (bodyType !== 'all') {
    bodies = bodies.filter((body) => (
      body.type === bodyType
    ));
  }
  return (
    <div className="list-container">
      {
        bodies.map((body) => (
          <>
            <Link
              className="body"
              key={body.id}
              data-after-content={body.name}
              to={`/${bodyType}s/${body.id}`}
              style={{
                background: `url(${body.picture}) center`,
                backgroundSize: 'cover',
              }}
            />
          </>
        ))
      }
    </div>
  );
}

const mapStateToProps = (state) => ({
  bodies: state.solarSystemReducer.bodies,
  planets: state.solarSystemReducer.planets,
  moons: state.solarSystemReducer.moons,
});

BodyList.propTypes = {
  bodyType: PropTypes.string.isRequired,
  bodies: PropTypes.instanceOf(Array).isRequired,
};

export default connect(mapStateToProps)(BodyList);
