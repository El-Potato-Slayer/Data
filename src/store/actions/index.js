const setBodies = (payload) => ({
  type: 'SET_BODIES',
  payload,
});

export default setBodies;
