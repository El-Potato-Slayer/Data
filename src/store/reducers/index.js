import { combineReducers } from 'redux';
import solarSystemReducer from './solarSystemReducer';
import filterReducer from './filterReducer';

export default combineReducers({ solarSystemReducer, filterReducer });
