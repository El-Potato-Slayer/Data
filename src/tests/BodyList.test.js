import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import store from '../store/configureStore';
import '@testing-library/jest-dom';
import BodyList from '../containers/BodyList';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <MemoryRouter>
        <BodyList bodyType="planets" />
      </MemoryRouter>
    </Provider>, div,
  );
});

it('renders correctly', () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <MemoryRouter initialEntries={['/']}>
        <BodyList bodyType="planets" />
      </MemoryRouter>
    </Provider>,
  );
  expect(getByTestId('list')).toHaveClass('list-container');
});

it('renders incorrectly', () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <MemoryRouter initialEntries={['/']}>
        <BodyList bodyType="planets" />
      </MemoryRouter>
    </Provider>,
  );
  expect(getByTestId('list')).not.toHaveClass('Uga');
});

it('renders snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <BodyList bodyType="planets" />
      </MemoryRouter>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
