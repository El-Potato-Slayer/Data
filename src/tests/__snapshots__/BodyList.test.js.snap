// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<div
  className="list-container"
  data-testid="list"
/>
`;
